playwright/tests/customers.spec.ts

```diff
@@ -0,0 +1,305 @@
+    newAddress.addressLine1,
+  );
+  await expect(
+    addressToVerify.getByTestId("postal-code-and-city"),
+  ).toContainText(` ${newAddress.zip} ${city}`);
+  await expect(
+    addressToVerify.getByTestId("country-area-and-country"),
+  ).toContainText(newAddress.country);
+  await addressesListPage.waitForDOMToFullyLoad();
+  await addressesListPage.clickShowMoreMenu(
+    CUSTOMERS.editCustomer.initialBillingAddress.lastName,
+  );
+  await addressesListPage.clickDeleteAddress();
+  await addressesListPage.deleteDialog.waitFor({ state: "visible" });
+  await customersPage.waitForNetworkIdle(() =>
+    deleteAddressDialog.clickDeleteButton(),
+  );
+  await expect(
+    addressesListPage.savedAddress.filter({
+      hasText: CUSTOMERS.editCustomer.initialBillingAddress.lastName,
+    }),
+  ).not.toBeVisible();
+  const additionalAddressCard = addressesListPage.addressCard.filter({
+    hasText: CUSTOMERS.editCustomer.additionalAddress.lastName,
+  });
+  await addressesListPage.clickShowMoreMenu(
+    CUSTOMERS.editCustomer.additionalAddress.lastName,
+  );
+  await addressesListPage.setAsDeafultShippingAddress();
+  await expect(
+    additionalAddressCard.locator(addressesListPage.addressTypeTitle),
+  ).toHaveText("Default Shipping Address");
+});
+
+test("TC: SALEOR_207 Issue a new gift card for the customer @e2e @customer", async () => {
+  const amount = faker.datatype.number(1000).toPrecision(2).toString();
+  await customersPage.waitForNetworkIdle(() =>
+    customersPage.gotoCustomerDetailsPage(CUSTOMERS.editCustomer.id),
+  );
+  await customersPage.waitForNetworkIdle(() =>
+    customersPage.clickIssueNewGiftCard(),
+  );
+  await customersPage.issueGiftCardDialog.typeAmount(amount);
+  await customersPage.issueGiftCardDialog.typeTag(faker.lorem.word());
+  await customersPage.issueGiftCardDialog.typeNote(faker.lorem.sentences(3));
+  await customersPage.waitForNetworkIdle(() =>
+    customersPage.issueGiftCardDialog.clickIssueButton(),
+  );
+  await customersPage.expectSuccessBanner();
+  await expect(giftCardsPage.issueGiftCardDialog.cardCode).toBeVisible();
+  const code = (
+    await giftCardsPage.issueGiftCardDialog.cardCode.innerText()
+  ).slice(-4);
+  await giftCardsPage.issueGiftCardDialog.clickCopyCodeButton();
+  await giftCardsPage.expectSuccessBanner();
+  await giftCardsPage.issueGiftCardDialog.clickOkButton();
+  await giftCardsPage.giftCardDialog.waitFor({ state: "hidden" });
+  await giftCardsPage.expectSuccessBannerMessage(
+    "Successfully created gift card",
+  );
+  await giftCardsPage.successBanner.first().waitFor({
+    state: "hidden",
+    timeout: 30000,
+  });
+  await giftCardsPage.waitForNetworkIdle(() =>
+    giftCardsPage.gotoGiftCardsListView(),
+  );
+  await giftCardsPage.waitForDOMToFullyLoad();
+  await giftCardsPage.gridCanvas
+    .getByText(`Code ending with ${code}`)
+    .waitFor({ state: "attached", timeout: 50000 });
+});
```

Comment:
Same here, a helper called waitForCanvasContainsCode (or sth like this ) would make it reusable across other tests and is cleaner

Reply:
waitForCanvasContainsText() added